### PR TITLE
feat: allow '*' wild card for tag option

### PR DESCRIPTION
**Pull request by EslamHiko**
This PR contains a:

- [ ] **bugfix**
- [x] new **feature**
- [ ] **code refactor**
- [x] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case
solve : https://github.com/webpack-contrib/html-loader/issues/267

### Breaking Changes
No

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -270,6 +270,51 @@ module.exports = {
...
+    rules: [
+      {
+        test: /\.html$/i,
+        loader: 'html-loader',
+        options: {
+          attributes: {
+            list: [
+              {
+                // Tag name
+                tag: '*', // or tag: ''
```
Comment (evilebottnawi) [inline README.md]: No need to support `tag: '*'`, if you don't provide `tag` try to use on all attributes, i.e `tag` can be `string` or `undefined` (not specify)

Comment (EslamHiko) [reply README.md]: It will work now when the string is empty or it's undefined

File: src/options.json [hunk 2]
```diff
@@ -5,8 +5,7 @@
       "type": "object",
       "properties": {
         "tag": {
-          "type": "string",
-          "minLength": 1
+          "type": "string"
```
Comment (evilebottnawi) [inline src/options.json]: Need to return it to `minLength` to `1`

File: src/plugins/source-plugin.js [hunk 3]
```diff
@@ -495,7 +495,13 @@ export default (options) =>
...
       return attributeList.find(
         (element) =>
-          element.tag.toLowerCase() === tag.toLowerCase() &&
+          // eslint-disable-next-line no-undefined
+          ((element.tag !== undefined &&
+            element.tag.toLowerCase() === tag.toLowerCase()) ||
+            // eslint-disable-next-line no-undefined
+            element.tag === undefined ||
+            // eslint-disable-next-line no-undefined
+            (element.tag !== undefined && !element.tag.length)) &&
```
Comment (evilebottnawi) [inline src/plugins/source-plugin.js]: I think we can simplify it

File: test/attributes-option.test.js [hunk 4]
```diff
@@ -172,6 +172,100 @@ describe("'attributes' option", () => {
...
+    expect(getWarnings(stats)).toMatchSnapshot('warnings');
+    expect(getErrors(stats)).toMatchSnapshot('errors');
+  });
+
+  it('should handle all src attributes in all HTML tags except img (testing filter option) tag is an empty string', async () => {
+    const compiler = getCompiler('simple.js', {
+      attributes: {
+        list: [
+          {
+            tag: '',
```
Comment (evilebottnawi) [inline test/attributes-option.test.js]: Should throw an error

File: test/attributes-option.test.js [hunk 5]
```diff
@@ -172,6 +172,76 @@ describe("'attributes' option", () => {
...
+    const stats = await compile(compiler);
+    try {
+      getModuleSource('./simple.html', stats);
+      execute(readAsset('main.bundle.js', compiler, stats));
+      expect(getWarnings(stats)).toMatchSnapshot('warnings');
+      expect(getErrors(stats)).toMatchSnapshot('errors');
+    } catch (e) {
+      expect(e.message.search('ValidationError')).not.toEqual(-1);
+    }
+  });
```
Comment (evilebottnawi) [inline test/attributes-option.test.js]: Move it to `validation.test.js`

File: src/plugins/source-plugin.js [hunk 6]
```diff
@@ -495,7 +495,11 @@ export default (options) =>
     const getAttribute = (tag, attribute, attributes, resourcePath) => {
       return attributeList.find(
         (element) =>
-          element.tag.toLowerCase() === tag.toLowerCase() &&
+          // eslint-disable-next-line no-undefined
+          (element.tag === undefined ||
+            // eslint-disable-next-line no-undefined
+            (element.tag !== undefined &&
```
Comment (evilebottnawi) [inline src/plugins/source-plugin.js]: Use `type of value === "undefined"` to avoid disable the rule

Comment (EslamHiko) [reply src/plugins/source-plugin.js]: Done :+1:

**Comment by evilebottnawi** [comment]
Thanks!

**Pull request merged**